features/data_converter/binary_protobuf/feature.ts

```diff
@@ -0,0 +1,59 @@
+import { Feature } from '@temporalio/harness';
+import * as proto from '@temporalio/proto';
+import * as assert from 'assert';
+
+// Inject Buffer and Uint8Array into isolate to workaround SDK bug
+// TODO(antlai-temporal) Remove when SDK bug is fixed
+const g = globalThis as any;
+g.Uint8Array = g.constructor.constructor('return globalThis.Uint8Array')();
+g.Buffer = g.constructor.constructor('return globalThis.Buffer')();
+
+const expectedResult = proto.temporal.api.common.v1.DataBlob.create({
+  encodingType: proto.temporal.api.enums.v1.EncodingType.ENCODING_TYPE_UNSPECIFIED,
+  data: new Uint8Array([0xde, 0xad, 0xbe, 0xef]),
+});
+
+// An "echo" workflow
+export async function workflow(
+  res: proto.temporal.api.common.v1.DataBlob
+): Promise<proto.temporal.api.common.v1.DataBlob> {
+  return res;
+}
+
+export const feature = new Feature({
+  workflow,
+  workflowStartOptions: {
+    args: [expectedResult],
+  },
+  dataConverter: {
+    payloadConverterPath: require.resolve('./binary_protobuf_converter'),
+  },
+  async checkResult(runner, handle) {
+    // verify client result is DataBlob `0xdeadbeef`
+    const result = await handle.result();
+    assert.deepEqual(result.toJSON(), expectedResult.toJSON());
```

Comment:
Chad's comment in the go version was not to compare protobuf directly, because it is a bit brittle, and use proto.Equal instead. There is no equivalent in protobufjs afaik, so I was just trying to make the check a bit more robust...

Reply:
I don't think that applies here, you can remove if deepEqual works as expected.